### DISCUSSION TITLE
Preprocessing images for template registration

### DISCUSSION THREAD

**Discussion by pnewstein** [navis-org/navis#144] [Q&A] [answered]
Thanks for building Navis! It's great to have quantitative morphology analysis accessible to the python ecosystem. Do you have advice for preprocessing images to prepare confocal brp stains for CMTK?

I looked at [Janelia’s flylight pipeline,](https://github.com/JaneliaSciComp/workstation/blob/master/docs/SamplePipeline.md) and it seems that [this imagej macro](https://github.com/JaneliaSciComp/jacs-tools/blob/master/aligner_jrc2018_20x_gen1/aligner/fiji_macros/20x_40x_Brain_Global_Aligner_Pipeline.ijm) determines whether optic lobes are present and perhaps modifies the image in various ways. I haven't been able to run the macro on my system yet. And instead have attempted to proceed with the CMTK commands found in [this shell script](https://github.com/JaneliaSciComp/jacs-tools/blob/master/aligner_jrc2018_20x_gen1/aligner/20xBrain_Align_CMTK.sh) without preprocessing. 

This approach has had some success. However, the CMTK commands ```make_initial_affine``` and ```registration``` often causes the brain to end up in the wrong orientation: flipped in the z or y axes. I have been able to work around this by flipping axes in the initial brp stains. I also found some success using landmarks to fit an initial affine, with the help of [xform.](https://github.com/schlegelp/xform)

My questions are:
- Is there a way to get brains into the right orientation that is more convenient than using landmarks?
- What are the best practices for manipulation of the raster images before feeding them to CMTK? For example, it appears that janelia’s macro manipulates the gamma of the input image.

**Comment by schlegelp** [comment]
Hi Peter. I have to admit my experience with CMTK is limited but I have asked another lab member. Will get back to you asap.

**Comment by scachero** [comment, marked as answer]
Hi Peter, 

This is Sebastian from the Jefferis lab. In general flipping axes works well for getting brain roughly aligned. Two things to keep in mind are that if you flip one axis then you change the handness of the brain (what was left now is right while dorsal remains dorsal and ventral remains ventral). Best practice in this case would be to change two axis to regain the original handness. 

Second thing is that brains which are imaged "upside down" tend to have drop off intensity opossite to the template and this creates issues for the registration which are not easy to overcome. As a general rule try to mount brains in the same way as the template you are registering to. Other than flipping, you can rotate the nrrd files in fiji. In general CMTK copes well with images which are less than 45 degrees out of alignment to the template. One thing to keep an eye on is that the image retains its calibration in um, otherwise cmtk gets confused.

In terms of manipulating the intensities of the images pre-registration it is my experience that that is not necessary. CMTK is pretty smart in equalising the image during the registration.

Last thing I would recommend if you are having trouble is to reformat the affine to check what is been fed to the warping registration. If the affine is not optimal then the warp is unlikely to work.

Hope this helps!

Best,

Sebastian Cachero

**Reply by pnewstein** [reply]
Thank you! I appreciate your help.
